cds/modules/deposit/static/js/cds_deposit/avc/components/cdsDeposit.js

```diff
@@ -804,6 +793,19 @@ function cdsDepositCtrl(
     that.setPristine();
   });
 
+  // Listen for any changes in the record state
+  $scope.$watch('$ctrl.record._cds.state', function(newVal, oldVal, scope) {
+    // The states have been changed
```

Comment:
Maybe add:
```
if (newVal !== oldVal) {
    ...
}
```
Ofter the watch is called multiple times for nothing... to be checked maybe with a log.